core-contracts/Loans/src/main/java/network/balanced/score/core/loans/LoansImpl.java

```diff
@@ -668,6 +676,48 @@ private void depositCollateral(String _symbol, BigInteger _amount, Address _from
         CollateralReceived(_from, _symbol, _amount);
     }
 
+    private void sellUserCollateral(Address from, BigInteger collateralToSell, String collateralSymbol, BigInteger minimumDebtToRepay) {
+        Context.require(collateralToSell.compareTo(BigInteger.ZERO) > 0, TAG + ": Sell amount must be more than zero.");
+        Context.require(PositionsDB.hasPosition(from), TAG + ": This address does not have a position on Balanced.");
+
+        Position position = PositionsDB.getPosition(from);
+
+        Context.require(position.getCollateral(collateralSymbol).compareTo(collateralToSell) >= 0, TAG + ": Position holds less " +
+                "collateral than the requested sell.");
+
+        String assetSymbol = BNUSD_SYMBOL;
+        Asset asset = AssetDB.getAsset(assetSymbol);
+        Address assetAddress = asset.getAssetAddress();
+
+        Collateral collateral = CollateralDB.getCollateral(collateralSymbol);
+        Address collateralAddress = collateral.getAssetAddress();
+
+        amountReceived.set(null);
+        BigInteger poolID = Context.call(BigInteger.class, dex.get(), "getPoolId", collateralAddress, assetAddress);
+        Context.require(poolID != null, TAG + ": There doesn't exist a bnUSD pool for "+ collateralSymbol + ".");
+
+        BigInteger userDebt = position.getDebt(collateralSymbol, assetSymbol);
+        Context.require(userDebt.compareTo(minimumDebtToRepay) >= 0, TAG + ": Minimum receive cannot be greater than your debt.");
+        expectedToken.set(assetAddress);
+
+        JsonObject swapData = Json.object();
+        swapData.add("method", "_swap");
+        swapData.add("params", Json.object().add("toToken", assetAddress.toString()).add("minimumReceive", minimumDebtToRepay.toString()));
+        byte[] data = swapData.toString().getBytes();
```

Comment:
Formatting here could be improved a little
```suggestion

        JsonObject swapParams = Json.object()
            .add("toToken", assetAddress.toString())
            .add("minimumReceive", minimumDebtToRepay.toString());
        JsonObject swapData = Json.object()
            .add("method", "_swap")
            .add("params", swapParams);
        byte[] data = swapData.toString().getBytes();
        
```